### ISSUE TITLE
Quick and easy model definition

### ISSUE THREAD

**Issue by gkourogiorgas** [uber/ludwig#115]
Hi, I have a dataset of 764 columns with numerical and binary values and I don't want to manually create the model definition dictionary. Is there a quick way to do it?

**Comment by athlonshi** [comment]
Write a Python code and create a dictionary for the input features with each column name. You can either use yaml package to save the dictionary to model definition file or use Ludwig API to take the dictionary as input.

**Comment by athlonshi** [comment]
Something like this
`import yaml
model_definition = {}
model_definition['input_features'] = [{'name': i, 'type': 'numerical'} for i in feature_columns]
model_definition['output_features'] = {}
model_definition['training'] = {}
ludwig_model = LudwigModel(model_definition)
`
Now your model_definition will merge with all other default settings, save it to a yaml file
use `yaml.dump(model_definition)`

**Comment by gkourogiorgas** [comment]
I created a data frame with column names: 'name' and 'type' with names and types respectively. How do I convert this data frame to the dictionary you want?
I tried df.to_dict but it didn't give what you wanted

**Comment by gkourogiorgas** [comment]
I used this:
`model_definition['input_features'] = [{'name': predictors.name[i], 'type': predictors.type[i]} for i in range(0,len(predictors))]`

**Comment by athlonshi** [comment]
This looks fine to me. What is wrong with it?
> I used this:
> `model_definition['input_features'] = [{'name': predictors.name[i], 'type': predictors.type[i]} for i in range(0,len(predictors))]`

**Comment by gkourogiorgas** [comment]
Not sure if it has to do with the model definition but I got an error:
`InvalidArgumentError: You must feed a value for placeholder tensor 'closefwd/closefwd_placeholder' with dtype float and shape [?]`

The code I ran is:
`model_definition = {}
model_definition['input_features'] = [{'name': predictors.name[i], 'type': predictors.type[i]} for i in range(0,len(predictors))]
model_definition['output_features'] = [{'name': 'closefwd', 'type': 'numerical'}]
model_definition['training'] = {}

model = LudwigModel(model_definition)
model.train(X_train)`

And 'closefwd' is included in X_train

**Comment by athlonshi** [comment]
try 
`ludwig_model.train(data_df=X_train)`?

**Comment by gkourogiorgas** [comment]
Same error

**Comment by athlonshi** [comment]
Great. Anything related to the input setup?

**Comment by gkourogiorgas** [comment]
yeah. I included the label to the features as well

**Comment by w4nderlust** [comment]
Thanks @athlonshi for your help! Closing as the issue seems solved.

**Comment by TimusLetap** [comment]
So the documentation provides that a model_definition.yaml file is required to train and test models in ludwig by Uber. I'm trying to use it on a custom dataset which I have a CSV file that was read into pandas as a dataframe as well. I tried to create a model_definition

I have been attempting to use this approach from github responses by ludwig folks:

import yaml

model_definition = {}
model_definition['input_features'] = [{'name': acoustic_data, type: category}, {'name': time_to_failure, type: numerical}]
model_definition['output_features'] = [{'name': time_to_failure.train.csv, 'type': numerical,loss:mean_absolute_error}]
model_definition['training'] = {}
ludwig_model = LudwigModel(model_definition)
train_stats = ludwig_model.train(data_df=LANL_data)
the documentation for how to create model_definitions needs to be more detailed.

**Comment by w4nderlust** [comment]
@TimusLetap not sure what your question is. The user guide, which explains how to create model definitions, is more than 100 pages long if you try to print it, it surely can be improved (like everything else) but I don't think it lacks detail. So either post a question or try to suggest improvements, otherwise your comments are pretty useless for us.

**Comment by TimusLetap** [comment]
Is there a template for YAML files to build on that is usable for custom datasets?

**Comment by TimusLetap** [comment]
> @TimusLetap not sure what your question is. The user guide, which explains how to create model definitions, is more than 100 pages long if you try to print it, it surely can be improved (like everything else) but I don't think it lacks detail. So either post a question or try to suggest improvements, otherwise your comments are pretty useless for us.

Because descriptions are recycled and revisited does not exactly make it detailed. It would be nice to know this so that contributions would be more fluid as model_definition.yaml is a key component to having it all function, if I'm not mistaken. I would be very appreciative to know a detailed template to customize from for custom uses.

**Comment by w4nderlust** [comment]
Recycled from what? I spent hours and hours writing everything myself, you are absolutely mistaken, cannot even imagine what you are talking about.
There are 12 examples in the documentation that you can copy-paste from which, with a minimum of imagination, can be combined in probably more than 50 different application examples, and, again, you already have 100 pages of user guide that explains everything, every parameter at your disposal and even goes as far as providing suggestions for what to use in which circumstance.
All this for a project that is one month old.
What I actually appreciate are users that are supportive of the work and that help concretely with PRs and useful suggestions. You haven't done any of those things so far.

**Comment by TimusLetap** [comment]
I apologise for my lack of understanding for applying it on a dataset. I am sorry I am not one of your ideal users for feedback. I do not mean that all the long hours and dedication you have put in place is not substantial. I (specific to me) was having difficulty in applying the concepts to a custom dataset. I found the examples super useful but they had some provided model_definition.yaml files to use. In regards to colab usage I saw that there were some errors, but some were in relation to the yaml file. I thought maybe there might be a better suggestion is all. Now I understand I just need to review more details to apply to my problem. I'll keep working at it!
